src/parsers/minecraft/coord/coord_base.ts

```diff
@@ -42,9 +51,9 @@ export class CoordBaseParser implements Parser {
                 );
             }
             switch (reader.peek()) {
-                case "~":
+                case RELATIVE:
                     if (hasLocal) {
-                        return helper.fail(
+                        helper.addErrors(
```

Comment:
Again, ideally this should only be an error if readNumber succeeds, which covers the whole range of the number.

Same thing for the other branches.